[BUG] extension method clashes with existing method on different class (javac only)

**Describe the bug**
I defined an extension method `Optional.stream()` (yea, I'm stuck with Java 8). This has the same signature as the already existing `Collection.stream()`, which should be perfectly fine.

Unfortunately it isn't. Now the already existing `Collection.stream()` doesn't work anymore. And I never even used this new extension method.

The problem only happens, when `Collection.stream()` is called inside of a Lambda that is passed to yet another extension method (`flattenTree()`):
```java
list.stream().flattenTree(p -> p.kids().stream()).collect(Collectors.toList());
                                              ^
```
Here the first `stream()` is irrelevant. The second `.stream()` shows the error, it is called on the `List` returned by `p.kids()`.

But for some reason, Lombok wants `p.kids()` to return an `Optional` in order to call `.stream()` on it:

Output of _javac_ (Eclipse is fine):
```shell
javac -cp lombok-1.18.26.jar prob/lems/ProblemWithJavacButNotEclipse3.java
prob/lems/ProblemWithJavacButNotEclipse3.java:56: error: method stream in class Extensions cannot be applied to given types;
        list.stream().flattenTree(p -> p.kids().stream()).collect(Collectors.toList());
                                                      ^
  required: Optional<T>
  found: List<Person>
  reason: cannot infer type-variable(s) T
    (argument mismatch; List<Person> cannot be converted to Optional<T>)
  where T is a type-variable:
    T extends Object declared in method <T>stream(Optional<T>)
```


It works when the lambda is passed to a normal non-extension method (`flatMap`):
```java
list.stream().flatMap(p -> p.kids().stream()).collect(Collectors.toList());
```

It also works when the inner .stream() is called on something whose type is known:
```java
list.stream().flattenTree(p -> Collections.<Person>emptyList().stream()).collect(Collectors.toList());
```

So specifying the type of the `p` parameter also fixes the problem:
```java
list.stream().flattenTree((Person p) -> p.kids().stream()).collect(Collectors.toList());
```

**To Reproduce**

See here for an example project:
- https://github.com/marc-guenther/lombok-problems
Type `make problem3` to trigger.

The relevant section of the code:
- https://github.com/marc-guenther/lombok-problems/blob/9b971e2b86b208ec6d9deb65b3e0ad3630ce0c59/prob/lems/ProblemWithJavacButNotEclipse3.java#L53-L62

**Expected behavior**
Two things:
- javac and Eclipse should show the same behaviour
- The example should compile of course

**Version info (please complete the following information):**
- Lombok version: 1.18.24
- Platform: javac 1.8.0_282, Eclipse 2019-03 (4.11.0)

**Additional context**
Contrary to #3334, the `suppressBaseMethods` option did not make any difference here. I have it set to `false` by default, but even setting it to `true` did not change anything.

--- a/src/core/lombok/javac/handlers/HandleExtensionMethod.java
+++ b/src/core/lombok/javac/handlers/HandleExtensionMethod.java
@@ -152,8 +152,10 @@ public class HandleExtensionMethod extends JavacAnnotationHandler<ExtensionMetho
 		
 		@Override
 		public Void visitMethodInvocation(final MethodInvocationTree tree, final Void p) {
-			super.visitMethodInvocation(tree, p);
+			scan(tree.getTypeArguments(), p);
+			scan(tree.getMethodSelect(), p);
 			handleMethodCall((JCMethodInvocation) tree);
+			scan(tree.getArguments(), p);
 			return null;
 		}
 		
@@ -181,6 +183,7 @@ public class HandleExtensionMethod extends JavacAnnotationHandler<ExtensionMetho
 			JCTree resolvedReceiver = resolution.get(receiver);
 			if (resolvedReceiver == null || resolvedReceiver.type == null) return;
 			Type receiverType = resolvedReceiver.type;
+			if (receiverType.isErroneous()) return;
 			
 			// Skip static method access
 			Symbol sym = null;

--- a/test/transform/resource/after-delombok/ExtensionMethodInLambda.java
+++ b/test/transform/resource/after-delombok/ExtensionMethodInLambda.java
@@ -1,0 +1,34 @@
+// version 8:
+import java.util.function.Function;
+
+public class ExtensionMethodInLambda {
+	public void testSimple() {
+		String test = "test";
+		test = ExtensionMethodInLambda.Extensions.map(test, s -> ExtensionMethodInLambda.Extensions.reverse(s));
+	}
+
+	public void testSameName() {
+		String test = "test";
+		test = ExtensionMethodInLambda.Extensions.map(test, s -> s.trim());
+	}
+
+	public void testArgumentOfInvalidMethod() {
+		String test = "test";
+		test.invalid(s -> s.reverse());
+	}
+
+
+	static class Extensions {
+		public static <T, R> R map(T value, Function<T, R> mapper) {
+			return mapper.apply(value);
+		}
+
+		public static String reverse(String string) {
+			return new StringBuilder(string).reverse().toString();
+		}
+
+		public static String trim(Integer integer) {
+			return "0";
+		}
+	}
+}

--- a/test/transform/resource/after-ecj/ExtensionMethodInLambda.java
+++ b/test/transform/resource/after-ecj/ExtensionMethodInLambda.java
@@ -1,0 +1,33 @@
+import java.util.function.Function;
+import lombok.experimental.ExtensionMethod;
+public @ExtensionMethod(value = ExtensionMethodInLambda.Extensions.class) class ExtensionMethodInLambda {
+  static class Extensions {
+    Extensions() {
+      super();
+    }
+    public static <T, R>R map(T value, Function<T, R> mapper) {
+      return mapper.apply(value);
+    }
+    public static String reverse(String string) {
+      return new StringBuilder(string).reverse().toString();
+    }
+    public static String trim(Integer integer) {
+      return "0";
+    }
+  }
+  public ExtensionMethodInLambda() {
+    super();
+  }
+  public void testSimple() {
+    String test = "test";
+    test = ExtensionMethodInLambda.Extensions.map(test, (<no type> s) -> ExtensionMethodInLambda.Extensions.reverse(s));
+  }
+  public void testSameName() {
+    String test = "test";
+    test = ExtensionMethodInLambda.Extensions.map(test, (<no type> s) -> s.trim());
+  }
+  public void testArgumentOfInvalidMethod() {
+    String test = "test";
+    test.invalid((<no type> s) -> s.reverse());
+  }
+}

--- a/test/transform/resource/before/ExtensionMethodInLambda.java
+++ b/test/transform/resource/before/ExtensionMethodInLambda.java
@@ -1,0 +1,35 @@
+// version 8:
+import java.util.function.Function;
+import lombok.experimental.ExtensionMethod;
+
+@ExtensionMethod(value = ExtensionMethodInLambda.Extensions.class)
+public class ExtensionMethodInLambda {
+	public void testSimple() {
+		String test = "test";
+		test = test.map(s -> s.reverse());
+	}
+	
+	public void testSameName() {
+		String test = "test";
+		test = test.map(s -> s.trim());
+	}
+	
+	public void testArgumentOfInvalidMethod() {
+		String test = "test";
+		test.invalid(s -> s.reverse());
+	}
+	
+	static class Extensions {
+		public static <T, R> R map(T value, Function<T, R> mapper) {
+			return mapper.apply(value);
+		}
+		
+		public static String reverse(String string) {
+			return new StringBuilder(string).reverse().toString();
+		}
+		
+		public static String trim(Integer integer) {
+			return "0";
+		}
+	}
+}

--- a/test/transform/resource/messages-delombok/ExtensionMethodInLambda.java.messages
+++ b/test/transform/resource/messages-delombok/ExtensionMethodInLambda.java.messages
@@ -1,0 +1,1 @@
+19 cannot find symbol symbol: method invalid((s)->s.reverse()) location: variable test of type java.lang.String

--- a/test/transform/resource/messages-ecj/ExtensionMethodInLambda.java.messages
+++ b/test/transform/resource/messages-ecj/ExtensionMethodInLambda.java.messages
@@ -1,0 +1,1 @@
+19 The method invalid((<no type> s) -> {}) is undefined for the type String

--- a/test/transform/resource/messages-idempotent/ExtensionMethodInLambda.java.messages
+++ b/test/transform/resource/messages-idempotent/ExtensionMethodInLambda.java.messages
@@ -1,0 +1,1 @@
+17 cannot find symbol symbol: method invalid((s)->s.reverse()) location: variable test of type java.lang.String

